web/client/hooks/useModulePlugins.js

```diff
@@ -119,14 +121,20 @@ function useModulePlugins({
 
     useEffect(() => {
         const store = getStore();
+        const urlQuery = url.parse(window.location.href, true).hash;
         if (store.storeManager) {
+            // some plugins like ContextCreator have edge-case of this scenario:
+            // there are two PluginsContainer components, whereas inner one load another list of plugins, like mapViewer
+            // on second step of context creator. To make it properly work, we need to check if router.location.path has
+            // changed and do not mute epics unless that's the case. Epics can be unmuted with no limitations though.
             Object.keys(pluginsCache).forEach((plugin) => {
-                if (!pluginsKeys.includes(plugin)) {
+                if (!pluginsKeys.includes(plugin) && urlQuery !== location) {
```

Comment:
can you add a unit test?

Reply:
Added two tests to cover cases when epics should be muted (route changed) and not.

Reply:
ty, can you check why pr is failing?

Reply:
Problem resolved